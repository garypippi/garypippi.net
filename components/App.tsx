import { PropsWithChildren } from 'react'
import 'tailwindcss/tailwind.css'

export const App = (props: PropsWithChildren<any>) => {
    return (
        <div className="container mx-auto">
            {props.children}
        </div>
    )
}
