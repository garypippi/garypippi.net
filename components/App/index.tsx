import { PropsWithChildren } from 'react'
import Link from 'next/link'
import { css } from 'goober'
import { theme } from '../../modules/csstheme'

export const App = ({ children }: PropsWithChildren<{}>) => {
    return (
        <>
            <div className={css`
                padding: 40px 0;
                display: flex;
                justify-content: center;
                align-items: center;
            `}>
                <Link
                    href="/"
                    className={css`
                        color: ${theme.color.title};
                `}>
                    {'garypippi.net'}
                </Link>
            </div>
            <div className={css`
                max-width: ${theme.breakpoint}px;
                margin: 0 auto;
                padding-bottom: 40px;
                @media screen and (max-width: ${theme.breakpoint}px) {
                    max-width: 100%;
                    padding-left: 10px;
                    padding-right: 10px;
                }
            `}>
                {children}
            </div>
        </>
    )
}
