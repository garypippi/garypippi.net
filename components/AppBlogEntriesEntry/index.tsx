import Link from 'next/link'
import { css } from 'goober'
import { color } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { format } from 'date-fns'

interface Props {
    entry: Omit<Entry, 'body'>
    className?: string
}

export const AppBlogEntriesEntry = ({
    className = '',
    entry: { href, attr },
}: Props) => {
    return (
        <div
            className={
                className +
                ' ' +
                css`
                    display: inline-block;
                    padding: 10px;
                `
            }
        >
            <Link
                href={href}
                className={css`
                    text-decoration: none;
                `}
            >
                <h3
                    className={css`
                        font-weight: normal;
                        font-size: 18px;
                        margin-bottom: 4px;
                        color: ${color['grey-5']};
                    `}
                >
                    {attr.title}
                </h3>
            </Link>
            <span
                className={css`
                    font-size: 16px;
                    color: ${color['grey-1']};
                `}
            >
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <Link
                    key={i}
                    href={`/tags/${tag}`}
                    className={css`
                        font-size: 16px;
                        margin-left: 4px;
                        color: ${color['grey-1']};
                    `}
                >
                    {`#${tag}`}
                </Link>
            ))}
        </div>
    )
}
