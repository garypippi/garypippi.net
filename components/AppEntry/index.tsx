import Link from 'next/link'
import { Entry } from '../../modules/markdown/types'
import { css } from 'goober'
import { theme } from '../../modules/csstheme'
import { format } from 'date-fns'

interface Props {
    entry: Entry
}

export const AppEntry = ({ entry: { href, attr } }: Props) => {
    return (
        <Link
            href={href}
            className={css`
                display: block;
                text-decoration: none;
                &:not(:last-child) {
                    margin-bottom: 16px;
                }
        `}>
            <h3 className={css`
                font-size: 20px;
                font-weight: normal;
                margin-bottom: 8px;
                color: ${theme.color.text};
            `}>
                {attr.title}
            </h3>
            <span className={css`
                font-size: 16px;
                color: ${theme.color.text};
            `}>
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <span
                    key={i}
                    className={css`
                    font-size: 14px;
                    padding: 2px 4px;
                    margin-left: 4px;
                    color: ${theme.color.text};
                    background-color: ${theme.color.tag};
                `}>
                    {tag}
                </span>
            ))}
        </Link>
    )
}
