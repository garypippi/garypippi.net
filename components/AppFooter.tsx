import { AppIcon } from './AppIcon'
import { useSnsLinks } from '../modules/store'

export const AppFooter = () => {
    return (
        <div className="2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4 my-8 py-8 text-gray-400">
            <div className="flex flex-row justify-center items-center">
                {useSnsLinks().map(({ href, icon }, i) => href && (
                    <a
                        key={`footer-sns-links-${i}`}
                        href={href}
                        className="mr-2"
                    >
                        <AppIcon
                            size={24}
                            className="inline"
                        >
                            {icon}
                        </AppIcon>
                    </a>
                ))}
            </div>
        </div>
    )
}
