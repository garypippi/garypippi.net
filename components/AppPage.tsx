import { PropsWithChildren } from 'react'

const classes = {
    md: '2xl:px-96 xl:px-64 lg:px-48 md:px-16 px-4',
    lg: '2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4'
}

interface Props {
    size: keyof (typeof classes)
}

export const AppPage = (props: PropsWithChildren<Props>) => {
    return (
        <div className={classes[props.size]}>
            {props.children}
        </div>
    )
}
