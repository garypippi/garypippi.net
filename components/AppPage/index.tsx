import { PropsWithChildren } from 'react'

interface ClassMap {
    md: string
    lg: string
}

const classes: ClassMap = {
    md: '2xl:px-96 xl:px-64 lg:px-48 md:px-16 px-4',
    lg: '2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4'
}

interface Props {
    size: keyof ClassMap
}

export const AppPage = ({ size, children }: PropsWithChildren<Props>) => {
    return (
        <div className={classes[size]}>
            {children}
        </div>
    )
}
