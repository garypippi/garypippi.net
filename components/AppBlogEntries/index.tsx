import { css } from 'goober'
import { lg, md } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { AppBlogEntriesEntry } from '../AppBlogEntriesEntry'

interface Props {
    entries: Omit<Entry,'body'>[]
}

export const AppBlogEntries = ({ entries }: Props) => {
    return (
        <div className={css`
            max-width: ${lg}px;
            margin: 0 auto;
        `}>
            {entries.map((entry, i) => (
                <AppBlogEntriesEntry
                    key={i}
                    entry={entry}
                    className={css`
                        width: calc(50% - 10px);
                        margin-bottom: 10px;
                        &:not(:nth-of-type(2n)) {
                            margin-right: 10px;
                        }
                `}/>
            ))}
        </div>
    )
}
