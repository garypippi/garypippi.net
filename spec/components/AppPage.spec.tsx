import renderer from 'react-test-renderer'
import { AppPage } from '../../components/AppPage'

describe('AppPage.tsx', () => {
    it('should render md', () => {
        expect(renderer.create(<AppPage size="md">hoge</AppPage>)).toMatchSnapshot()
    })
    it('should render lg', () => {
        expect(renderer.create(<AppPage size="lg">hoge</AppPage>)).toMatchSnapshot()
    })
})
