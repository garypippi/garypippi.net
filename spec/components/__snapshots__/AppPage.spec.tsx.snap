// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPage.tsx should render lg 1`] = `"<div class="2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4">hoge</div>"`;

exports[`AppPage.tsx should render md 1`] = `"<div class="2xl:px-96 xl:px-64 lg:px-48 md:px-16 px-4">hoge</div>"`;
